jerry-core/ecma/builtin-objects/typedarray/ecma-builtin-typedarray-prototype.c

```diff
@@ -41,6 +41,53 @@
 #define ECMA_BUILTINS_INTERNAL
 #include "ecma-builtins-internal.h"
 
+/**
+ * This object has a custom dispatch function.
+ */
+#define BUILTIN_CUSTOM_DISPATCH
+
+/**
+ * List of built-in routine identifiers.
+ */
+enum
+{
+  /* These routines must be in this order */
+  ECMA_TYPEDARRAY_PROTOTYPE_ROUTINE_START = ECMA_BUILTIN_ID__COUNT - 1,
+  ECMA_TYPEDARRAY_PROTOTYPE_ROUTINE_MAP,
+  ECMA_TYPEDARRAY_PROTOTYPE_ROUTINE_REDUCE,
+  ECMA_TYPEDARRAY_PROTOTYPE_ROUTINE_REDUCE_RIGHT,
+  ECMA_TYPEDARRAY_PROTOTYPE_ROUTINE_FIND,
+  ECMA_TYPEDARRAY_PROTOTYPE_ROUTINE_FIND_INDEX,
+  ECMA_TYPED_ARRAY_PROTOTYPE_INCLUDES,
+
+  ECMA_TYPEDARRAY_PROTOTYPE_ROUTINE_EVERY,
+  ECMA_TYPEDARRAY_PROTOTYPE_ROUTINE_SOME,
+  ECMA_TYPEDARRAY_PROTOTYPE_ROUTINE_FOR_EACH,
+  ECMA_TYPEDARRAY_PROTOTYPE_ROUTINE_FILTER,
+
+  ECMA_TYPEDARRAY_PROTOTYPE_ROUTINE_BYTELENGTH_GETTER,
+  ECMA_TYPEDARRAY_PROTOTYPE_ROUTINE_BYTEOFFSET_GETTER,
+  ECMA_TYPEDARRAY_PROTOTYPE_ROUTINE_LENGTH_GETTER,
+
+  ECMA_TYPEDARRAY_PROTOTYPE_ROUTINE_BUFFER_GETTER,
+  ECMA_TYPEDARRAY_PROTOTYPE_ROUTINE_REVERSE,
+  ECMA_TYPEDARRAY_PROTOTYPE_ROUTINE_SLICE,
+  ECMA_TYPEDARRAY_PROTOTYPE_ROUTINE_COPY_WITHIN,
+  ECMA_TYPEDARRAY_PROTOTYPE_ROUTINE_SUBARRAY,
+  ECMA_TYPEDARRAY_PROTOTYPE_ROUTINE_INDEX_OF,
+  ECMA_TYPEDARRAY_PROTOTYPE_ROUTINE_LAST_INDEX_OF,
+  ECMA_TYPEDARRAY_PROTOTYPE_ROUTINE_TO_LOCALE_STRING,
+  ECMA_TYPEDARRAY_PROTOTYPE_ROUTINE_KEYS,
+  ECMA_TYPEDARRAY_PROTOTYPE_ROUTINE_ENTRIES,
+  ECMA_TYPEDARRAY_PROTOTYPE_ROUTINE_SET,
+  ECMA_TYPEDARRAY_PROTOTYPE_ROUTINE_FILL,
+  ECMA_TYPEDARRAY_PROTOTYPE_ROUTINE_SORT,
+
+  ECMA_TYPEDARRAY_PROTOTYPE_ROUTINE_TO_STRING_TAG_GETTER,
+  ECMA_TYPEDARRAY_PROTOTYPE_ROUTINE_TO_STRING,
+  ECMA_TYPEDARRAY_PROTOTYPE_ROUTINE_JOIN,
```

Comment:
Several of these methods should also throw if their 'this' value is not a typedarray.